@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -749,65 +751,69 @@ export class TiltViewClipPlanesEvaluator extends TopViewClipPlanesEvaluator {
-
-            // Ratio of the depth of the camera compared to the distance between the center of the
-            // screen and the middle of the top of the screen (here set to be value 1). We just need
-            // this to be a ratio because we are interested in just computing the direction of the
-            // upper right corner and not any specific length.
-            const zLength = 1 / Math.tan(halfVerticalFovAngle);
-
-            // Camera space direction vector along the top right frustum edge (of the camera).
-            const upperRightDirection = this.m_tmpVectors[0]
-                .set(camera.aspect, 1, -zLength)
-                .applyMatrix4(camera.matrixWorld)
-                .sub(camera.position);
-
-            // Ray must be given normalized vector, otherwise it won't work
-            this.m_ray.set(camera.position, upperRightDirection.normalize());
-
-            // World space intersection with sphere or null
-            const fovSphereIntersection = this.m_ray.intersectSphere(
-                this.m_sphere,
-                this.m_tmpVectors[0]
-            );
+    private computeNearDistSphericalProj(
+        camera: THREE.PerspectiveCamera,
+        projection: Projection
+    ): number {
+        assert(projection.type === ProjectionType.Spherical);
 
-            if (fovSphereIntersection === null) {
-                // Use tangent based far plane if horizon is within field of view
-                viewRanges.far = SphericalProj.getFarDistanceFromElevatedHorizon(
-                    camera,
-                    d,
-                    r,
-                    maxR
-                );
-            } else {
-                viewRanges.far = SphericalProj.getFarPlaneBasedOnFovIntersection(
-                    fovSphereIntersection,
-                    camera
-                );
-            }
-        } else {
-            viewRanges.far = this.getOrthoBasedFarPlane(d, r);
+        // Default near plane approximation.
+        const defaultNear = projection.groundDistance(camera.position) - this.maxElevation;
+        const cameraBelowMaxElevation = defaultNear <= 0;
+        if (cameraBelowMaxElevation) {
+            // Near distance will be adjusted by constraints later.
+            return 0;
         }
 
-        return this.applyViewRangeConstraints(viewRanges, camera, projection, elevationProvider);
+        const perspectiveCam = camera as THREE.PerspectiveCamera;
+        const halfVerticalFovAngle = THREE.MathUtils.degToRad(perspectiveCam.fov / 2);
+        const maxR = EarthConstants.EQUATORIAL_RADIUS + this.maxElevation;
+        const ndcBottomDir = TiltViewClipPlanesEvaluator.NDC_BOTTOM_DIR;
+
+        // Take as near distance the distance to the point where the near plane is tangent to the
+        // globe. If the tangent point is not visible, take instead the distance to the intersection
+        // of the bottom frustum side's median and the globe.
```

Comment:
ditto, what does the `bottom frustum side's median` mean?

Reply:
removed.